File: interpreter-for-cs/src/Runtime/Sentence/SentenceImport.cs
/*
	PeakScript: https://github.com/Jenocn/PeakScript
	By Jenocn: https://jenocn.github.io
*/

namespace peak.interpreter {

	public class SentenceImport : Sentence {
		private string _moduleName = "";
		public SentenceImport(string moduleName) {
			_moduleName = moduleName;
		}
		public override ExecuteResult Execute(Space space) {
			var module = ModulePool.instance.UseModule(_moduleName);
			if (module == null) {
				ErrorLogger.LogRuntimeError(_moduleName);
				ErrorLogger.LogRuntimeError(ErrorRuntimeCode.Import, "The module \"" + _moduleName + "\" import failed!");
				return ExecuteResult.Failed;
			}
			if (!space.UseModule(module)) {
				ErrorLogger.LogRuntimeError(_moduleName);
				ErrorLogger.LogRuntimeError(ErrorRuntimeCode.Import, "The module \"" + _moduleName + "\" import failed!");
				return ExecuteResult.Failed;
			}
			return ExecuteResult.Successed;
		}

	}

} // namespace peak.interpreter

File: interpreter-for-cpp/src/Runtime/Sentence/SentenceImport.cpp
#include "SentenceImport.h"
#include "../Module.h"
#include "../ModulePool.h"

using namespace peak::interpreter;

SentenceImport::SentenceImport(const std::string& moduleName)
	: _moduleName(moduleName) {
}
ExecuteResult SentenceImport::Execute(std::shared_ptr<Space> space) {
	auto module = ModulePool::GetInstance()->UseModule(_moduleName);
	if (!module) {
		ErrorLogger::LogRuntimeError(_moduleName);
		ErrorLogger::LogRuntimeError(ErrorRuntimeCode::Import, "The module \"" + _moduleName + "\" import failed!");
		return ExecuteResult::Failed;
	}
	if (!space->UseModule(module)) {
		ErrorLogger::LogRuntimeError(_moduleName);
		ErrorLogger::LogRuntimeError(ErrorRuntimeCode::Import, "The module \"" + _moduleName + "\" import failed!");
		return ExecuteResult::Failed;
	}
	return ExecuteResult::Successed;
}